edgraph/server.go

```diff
@@ -1512,6 +1519,75 @@ func processQuery(ctx context.Context, qc *queryContext) (*api.Response, error)
 		qc.valRes[name] = v.Vals
 	}
 
+	for gmuIndex, gmu := range qc.gmuList {
+		for rdfIndex, pred := range gmu.Set {
+			condVar := qc.condVars[gmuIndex]
+			uids, isThere := qc.uidRes[condVar]
+
+			uniqueQueryVar := qc.uniqueVars[strconv.Itoa(gmuIndex)+strconv.Itoa(rdfIndex)]
+			uniqueQueryResult, ok := qr.Vars[uniqueQueryVar]
+
+			if !ok || !(isThere && len(uids) == 1) {
```

Comment:
I think do it in small steps, instead of a big condition. It would be difficult to read this way.